### PR TITLE
[Heartbeat] Use go-lookslike instead of mapval for heartbeat tests

### PR DESCRIPTION
**Pull request by andrewvc**
This switches our usage the built-in `mapval` package out in favor of the open-sourced [go-lookslike](https://github.com/elastic/go-lookslike) package in heartbeat. There are some other places where this is used, but this is a good starting point.

We can eventually delete mapval in a subsequent PR.

### CODE REVIEW

File: heartbeat/hbtest/hbtestutil.go [hunk 1]
```diff
@@ -30,10 +30,14 @@ import (
 	"strings"
 	"testing"
 
+	"github.com/elastic/go-lookslike/lookslike/isdef"
+	"github.com/elastic/go-lookslike/lookslike/validator"
+
 	"github.com/stretchr/testify/require"
 
+	"github.com/elastic/go-lookslike/lookslike"
```
Comment (ruflin) [inline heartbeat/hbtest/hbtestutil.go]: Import fun :-)

File: heartbeat/monitors/active/http/http_test.go [hunk 2]
```diff
@@ -31,14 +31,20 @@ import (
 	"testing"
 	"time"
 
+	"github.com/elastic/go-lookslike/lookslike/isdef"
+	"github.com/elastic/go-lookslike/lookslike/validator"
+
+	"github.com/elastic/go-lookslike/lookslike/testslike"
+
+	"github.com/elastic/go-lookslike/lookslike"
```
Comment (ruflin) [inline heartbeat/monitors/active/http/http_test.go]: Import fun. I think these should also go under the elastic block

File: vendor/vendor.json [hunk 3]
```diff
@@ -898,6 +898,48 @@
 			"version": "v0.4.0",
 			"versionExact": "v0.4.0"
 		},
+		{
```
Comment (ruflin) [inline vendor/vendor.json]: Should we push for a first release to have here a version linked?

**Comment by andrewvc** [comment]
@ruflin I've addressed all your comments!

**Comment by andrewvc** [comment]
jenkins, please retest this

**Comment by andrewvc** [comment]
jenkins, retest this please

**Comment by andrewvc** [comment]
jenkins, retest this please

**Comment by andrewvc** [comment]
Jenkins, retest this please

**Comment by andrewvc** [comment]
jenkins, retest this please

**Comment by andrewvc** [comment]
jenkins, retest this please

**Comment by andrewvc** [comment]
Filebeat failure on travis is unrelated

**Pull request merged**